scio-hdfs/src/main/scala/com/spotify/scio/hdfs/package.scala

```diff
@@ -158,16 +155,35 @@ package object hdfs {
       }
     }
 
-    private[scio] def hadoopDistCacheCopy(src: Path, target: URI, conf: Configuration): Unit = {
-      logger.debug(s"Will copy ${src.toUri}, to $target")
+    private def getTargetFromOptions() = {
+      if (self.options.getRunner() != classOf[DirectPipelineRunner]) {
+        val dfOptions = self.optionsAs[DataflowPipelineOptions]
+        require(self.optionsAs[DataflowPipelineOptions].getStagingLocation != null,
```

Comment:
Oh nice, so we can remove the require and just manually check `tempLocation` and `stagingLocation`?